Mathlib/Analysis/InnerProductSpace/Laplace.lean

```diff
@@ -0,0 +1,181 @@
+  fun x ↦ (fderiv 𝕜 (fderiv 𝕜 f) x).toLinearMap₂
+
+/--
+Expression of `bilinear_of_iteratedFDeriv_two` in terms of `iteratedFDeriv`.
+-/
+lemma bilinear_of_iteratedFDeriv_two_eq_iteratedFDeriv (f : E → F) (e e₁ e₂ : E) :
+    bilinear_of_iteratedFDeriv_two 𝕜 f e e₁ e₂ = iteratedFDeriv 𝕜 2 f e ![e₁, e₂] := by
+  simp [iteratedFDeriv_two_apply f e ![e₁, e₂], bilinear_of_iteratedFDeriv_two]
+
+variable (𝕜) in
+/--
+Convenience reformulation of the second iterated derivative, as a map from `E` to linear maps
+`E ⊗[𝕜] E →ₗ[𝕜] F`.
+-/
+noncomputable def tensor_of_iteratedFDeriv_two (f : E → F) : E → E ⊗[𝕜] E →ₗ[𝕜] F :=
+  fun e ↦ lift (bilinear_of_iteratedFDeriv_two 𝕜 f e)
+
+/--
+Expression of `tensor_of_iteratedFDeriv_two` in terms of `iteratedFDeriv`.
+-/
+lemma tensor_of_iteratedFDeriv_two_eq_iteratedFDeriv (f : E → F) (e e₁ e₂ : E) :
+    tensor_of_iteratedFDeriv_two 𝕜 f e (e₁ ⊗ₜ[𝕜] e₂) = iteratedFDeriv 𝕜 2 f e ![e₁, e₂] := by
+  rw [← bilinear_of_iteratedFDeriv_two_eq_iteratedFDeriv, tensor_of_iteratedFDeriv_two]
+  rfl
+
+end secondDerivativeAPI
+
+/-!
+## Definition of the Laplace Operator
+-/
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [InnerProductSpace ℝ E] [FiniteDimensional ℝ E]
+  {F : Type*} [NormedAddCommGroup F] [NormedSpace ℝ F]
+  {G : Type*} [NormedAddCommGroup G] [NormedSpace ℝ G]
+  {f f₁ f₂ : E → F} {x : E}
+
+variable (f) in
+/--
+Definition of the Laplace operator for functions on real inner product spaces.
+-/
+noncomputable def Real.Laplace : E → F :=
+  fun x ↦ tensor_of_iteratedFDeriv_two ℝ f x (InnerProductSpace.canonicalCovariantTensor E)
+
+/--
+Introduce `Δ` as a notation for the Laplace operator.
+-/
+notation "Δ" => Real.Laplace
+
+/-!
+## Computation of Δ in Terms of Orthonormal Bases
+-/
+
+variable (f) in
+/--
+Standard formula, computing the Laplace operator from any orthonormal basis.
+-/
+theorem laplace_eq_iteratedFDeriv_orthonormalBasis {ι : Type*} [Fintype ι]
+    (v : OrthonormalBasis ι ℝ E) :
+    Δ f = fun x ↦ ∑ i, iteratedFDeriv ℝ 2 f x ![v i, v i] := by
+  ext x
+  simp [Real.Laplace, canonicalCovariantTensor_eq_sum E v,
+    tensor_of_iteratedFDeriv_two_eq_iteratedFDeriv]
+
+variable (f) in
+/--
+Standard formula, computing the Laplace operator from the standard orthonormal basis of a real inner
+product space.
+-/
+theorem laplace_eq_iteratedFDeriv_stdOrthonormalBasis :
+    Δ f = fun x ↦
+      ∑ i, iteratedFDeriv ℝ 2 f x ![(stdOrthonormalBasis ℝ E) i, (stdOrthonormalBasis ℝ E) i] :=
+  laplace_eq_iteratedFDeriv_orthonormalBasis f (stdOrthonormalBasis ℝ E)
+
+/--
+Special case of the standard formula for functions on `ℂ`, with the standard structure as a real
+inner product space.
+-/
+theorem laplace_eq_iteratedFDeriv_complexPlane (f : ℂ → F) :
+    Δ f = fun x ↦
+      iteratedFDeriv ℝ 2 f x ![1, 1] + iteratedFDeriv ℝ 2 f x ![Complex.I, Complex.I] := by
+  simp [laplace_eq_iteratedFDeriv_orthonormalBasis f Complex.orthonormalBasisOneI]
+
+/-!
+## Congruence Lemma for Δ
+-/
```

Comment:
(I'm not sure if this intermediate heading is strictly necessary, but I won't object to it either.)

Reply:
Then I'd say: let's stick to it. Otherwise, the congruence lemma would end in the section "Computation of Δ in Terms of Orthonormal Bases".